modules/apm/src/main/java/METERING.md

```diff
@@ -0,0 +1,171 @@
+```
+
+If we don’t have access to ‘state’ that will be fetched on metric event (when callback is executed)
+we can use a utility LongGaugeMetric or LongGaugeMetric
+```java
+MeterRegistry meterRegistry ;
+LongGaugeMetric longGaugeMetric = LongGaugeMetric.create(meterRegistry, "es.test.gauge", "a test gauge", "total value");
+longGaugeMetric.set(123L);
+```
+### The use of attributes aka dimensions
+Each instrument can attach attributes to a reported value. This helps drilling down into the details
+of value that was reported during the metric event
+
+
+## Development
+
+### Fake http server
+
+The quickest way to verify that your metrics are working is to run `./gradlew run --with-apm-server`.
+This will run ES node (or nodes in serverless) and also start a fake http server that will act
+as an apm server. This fake http server will log all the http messages it receives from apm-agent
+
+### With APM server in cloud
+You can also run local ES node with an apm server in cloud.
+Create a new deployment in cloud, then click the 'hamburger :)' on the left, scroll to Observability and click APM under it.
+At the upper right corner there is `Add data` link, then scroll down to `ApmAgents` section and pick Java
+There you should be able to see `elastic.apm.secret_token` and `elastic.apm.server_url. You will use them in the next step.
+
+Next you should create a file `apm_server_ess.gradle`
+in a different directory than your elasticsearch checkout (so that branch changes don't remove it)
+The content of the file:
+```
+rootProject {
+  if (project.name == 'elasticsearch') {
+    afterEvaluate {
+      testClusters.matching { it.name == "runTask" }.configureEach {
+        setting 'xpack.security.audit.enabled', 'true'
+        keystore 'tracing.apm.secret_token', 'REDACTED'
+        setting 'telemetry.metrics.enabled', 'true'
+
+      setting 'tracing.apm.agent.server_url', 'https://REDACTED:443'
+      }
+    }
+  }
+}
+```
+Use the secret_token and server_url (REDACTED) from previous step.
+
+you can run your local ES node with APM in ESS with this command
+`./gradlew run -I ../apm_enable_statefull.gradle`
+
+#### An init.d gradle setup
+
+Alternatively you can edit your `~/.gradle/init.d/apm.gradle`
+```groovy
+rootProject {
+    if (project.name == 'elasticsearch' && Boolean.getBoolean('metrics.enabled')) {
+        afterEvaluate {
+            testClusters.matching { it.name == "runTask" }.configureEach {
+                setting 'xpack.security.audit.enabled', 'true'
+                keystore 'tracing.apm.secret_token', 'TODO-REPLACE'
+                setting 'telemetry.metrics.enabled', 'true'
+                setting 'tracing.apm.agent.server_url', 'https://TODO-REPLACE-URL.apm.eastus2.staging.azure.foundit.no:443'
+            }
+        }
+    }
+}
+```
+
+The example use:
+```
+./gradlew :run -Dmetrics.enabled=true
+```
+
+
+
+
+#### Logging
+with any approach you took to run your ES with APM you will find apm-agent.json file
```

Comment:
The `apm-agent.json` log file will be in the ES logs directory, eg `build/testclusters/runTask-0/logs/apm-agent.json`.

Reply:
yes, that is correct for ./gradlew run
I will skip the exact path, as in serverless you might have more clusters and when running Integration Tests the path will also be slightly different.
I think ES developers will be able to find where the ES logs dir is depending on how they run it